composeApp/src/commonMain/kotlin/world/respect/app/viewmodel/lessons/list/LessonListViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package world.respect.app.viewmodel.lessons.list
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.toRoute
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.jetbrains.compose.resources.getString
+import respect.composeapp.generated.resources.Res
+import respect.composeapp.generated.resources.lesson_list
+import world.respect.app.app.LessonDetail
+import world.respect.app.app.LessonList
+import world.respect.app.appstate.AppBarSearchUiState
+import world.respect.app.datasource.fakeds.FakeOpdsDataSource
+import world.respect.app.viewmodel.RespectViewModel
+import world.respect.datasource.DataLoadParams
+import world.respect.datasource.DataLoadResult
+import world.respect.datasource.opds.model.OpdsFacet
+import world.respect.datasource.opds.model.OpdsPublication
+import world.respect.datasource.opds.model.ReadiumLink
+import world.respect.navigation.NavCommand
+
+data class LessonListUiState(
+    val publications: List<OpdsPublication> = emptyList(),
+    val lessonFilter: List<OpdsFacet> = emptyList(),
+    val selectedFilterTitle: String? = null,
+    val link: List<ReadiumLink> = emptyList(),
+
+    )
+
+class LessonListViewModel(
+    savedStateHandle: SavedStateHandle
+) : RespectViewModel(savedStateHandle) {
+
+    private val opdsDataSource: FakeOpdsDataSource = FakeOpdsDataSource()
```

Comment:
This is **_NOT_** following the pattern used as I put in AppLauncherViewModel and AppListViewModel. You need to add ```RespectAppDataSourceProvider``` to the constructor parameters as per those viewmodels. Then get the datasource from that. 

This is using dependency injection (see ```AppKoinModule``` - this is important to ensure that the real data source can easily replace the fake data source app wide.

Please update all viewmodels accordingly.